src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/incidents/QuestionnaireResourceIntTest.kt

```diff
@@ -0,0 +1,234 @@
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access forbidden with wrong role`() {
+      webTestClient.get().uri("/questionnaires/ids")
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `access unauthorised with no auth token`() {
+      webTestClient.get().uri("/questionnaires/ids")
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+  }
+
+  @Nested
+  @DisplayName("GET /questionnaires/ids")
+  inner class GetQuestionnaireIds {
+    @Test
+    fun `get all question ids - no filter specified`() {
+      webTestClient.get().uri("/questionnaires/ids")
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$.numberOfElements").isEqualTo(3)
+    }
+
+    @Test
+    fun `can request a different page size`() {
+      webTestClient.get().uri {
+        it.path("/questionnaires/ids")
+          .queryParam("size", "2")
+          .build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("totalElements").isEqualTo(3)
+        .jsonPath("numberOfElements").isEqualTo(2)
+        .jsonPath("number").isEqualTo(0)
+        .jsonPath("totalPages").isEqualTo(2)
+        .jsonPath("size").isEqualTo(2)
+    }
+
+    @Test
+    fun `can request a different page`() {
+      webTestClient.get().uri {
+        it.path("/questionnaires/ids")
+          .queryParam("size", "2")
+          .queryParam("page", "1")
+          .build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("totalElements").isEqualTo(3)
+        .jsonPath("numberOfElements").isEqualTo(1)
+        .jsonPath("number").isEqualTo(1)
+        .jsonPath("totalPages").isEqualTo(2)
+        .jsonPath("size").isEqualTo(2)
+    }
+  }
+
+  @Nested
+  @DisplayName("GET /questionnaires/{questionnaireId}")
+  inner class GetQuestionnaire {
+
+    @Test
+    fun `will return the questionnaire by Id`() {
+      webTestClient.get().uri {
+        it.path("/questionnaires/${questionnaire1.id}")
+          .build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_INCIDENTS")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("id").isEqualTo(questionnaire1.id)
+        .jsonPath("code").isEqualTo("ESCAPE_EST")
+        .jsonPath("active").isEqualTo(true)
+        .jsonPath("description").isEqualTo("Escape Questionnaire")
+    }
+
+    @Test
+    fun `will return the questions and answers for a questionnaire`() {
+      val persistedQuestionnaire1 = webTestClient.get().uri {
```

Comment:
BTW you could sorten that to .get("/questionnaires/${questionnaire1.id}")) in this instance